edgelet/kube-client/src/client.rs

```diff
@@ -343,6 +342,159 @@ where
+            ..api_core::ListNamespacedServiceAccountOptional::default()
+        };
+
+        api_core::ServiceAccount::list_namespaced_service_account(namespace, params)
+            .map_err(Error::from)
+            .map(|req| {
+                self.request(req).and_then(|response| match response {
+                    api_core::ListNamespacedServiceAccountResponse::Ok(service_account_list) => {
+                        Ok(service_account_list)
+                    }
+                    _ => Err(Error::from(ErrorKind::Response)),
+                })
+            })
+            .into_future()
+            .flatten()
+    }
+
+    pub fn create_service_account(
+        &mut self,
+        namespace: &str,
+        service_account: &api_core::ServiceAccount,
+    ) -> impl Future<Item = api_core::ServiceAccount, Error = Error> {
+        api_core::ServiceAccount::create_namespaced_service_account(
+            namespace,
+            &service_account,
+            api_core::CreateNamespacedServiceAccountOptional::default(),
+        )
+        .map_err(Error::from)
+        .map(|req| {
+            self.request(req).and_then(|response| match response {
+                api_core::CreateNamespacedServiceAccountResponse::Accepted(service_account)
+                | api_core::CreateNamespacedServiceAccountResponse::Created(service_account)
+                | api_core::CreateNamespacedServiceAccountResponse::Ok(service_account) => {
+                    Ok(service_account)
+                }
+                _ => Err(Error::from(ErrorKind::Response)),
+            })
+        })
+        .into_future()
+        .flatten()
+    }
+
+    pub fn replace_service_account(
+        &mut self,
+        namespace: &str,
+        name: &str,
+        service_account: &api_core::ServiceAccount,
+    ) -> impl Future<Item = api_core::ServiceAccount, Error = Error> {
+        api_core::ServiceAccount::replace_namespaced_service_account(
+            name,
+            namespace,
+            service_account,
+            api_core::ReplaceNamespacedServiceAccountOptional::default(),
+        )
+        .map_err(Error::from)
+        .map(|req| {
+            self.request(req).and_then(|response| match response {
+                api_core::ReplaceNamespacedServiceAccountResponse::Created(service_account)
+                | api_core::ReplaceNamespacedServiceAccountResponse::Ok(service_account) => {
+                    Ok(service_account)
+                }
+                _ => Err(Error::from(ErrorKind::Response)),
+            })
+        })
+        .into_future()
+        .flatten()
+    }
+
+    pub fn list_role_binding(
+        &mut self,
+        _namespace: &str,
+        name: Option<&str>,
+        label_selector: Option<&str>,
+    ) -> impl Future<Item = rbac::ClusterRoleBindingList, Error = Error> {
+        let field_selector =
+            name.and_then(|role_binding| Some(format!("metadata.name={}", role_binding)));
```

Comment:
Should be able to simply with a `map` instead of `and_then` followed by `Some` -- like I commented earlier for the `list_service_accounts` function.